codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoStackStepMiddlewareGenerator.java

```diff
@@ -64,11 +68,13 @@ public static GoStackStepMiddlewareGenerator createInitalizeStepMiddleware(Strin
     /**
      * Create a new BuildStep middleware generator with the provided type name.
      *
-     * @param identifier is the type name to identify the middleware.
+     * @param name is the type name to identify the middleware.
+     * @param id   the unique ID for the middleware.
      * @return the middleware generator.
      */
-    public static GoStackStepMiddlewareGenerator createBuildStepMiddleware(String identifier) {
-        return createMiddleware(identifier,
+    public static GoStackStepMiddlewareGenerator createBuildStepMiddleware(String name, String id) {
+        return createMiddleware(name,
+                id,
```

Comment:
How will this behavior be replicated for non generated middlewares? i.e. making sure they always have an id too. 

Should we have an interface contract to make sure middlewares have an id defined.

Reply:
Non-generated middleware will have to be set when they are defined. That is outside of the scope of this since this making the generated middleware bit more generic.

Reply:
Also sorry forgot to answer the first part of your question. The `ID` is an expected member of the smithy Middleware interface types, https://github.com/awslabs/smithy-go/blob/master/middleware/step_build.go#L28-L39